pkg/sentry/socket/epsocket/stack.go

```diff
@@ -144,7 +144,98 @@ func (s *Stack) SetTCPSACKEnabled(enabled bool) error {
+		out := Metrics.ICMP.V4PacketsSent.ICMPv4PacketStats
+		*stats = inet.StatSNMPICMP{
+			0, // InMsgs
+			Metrics.ICMP.V4PacketsSent.Dropped.Value(), // InErrors
+			0, // InCsumErrors
+			in.DstUnreachable.Value(), // InDestUnreachs
+			in.TimeExceeded.Value(),   // InTimeExcds
+			in.ParamProblem.Value(),   // InParmProbs
+			in.SrcQuench.Value(),      // InSrcQuenchs
+			in.Redirect.Value(),       // InRedirects
+			in.Echo.Value(),           // InEchos
+			in.EchoReply.Value(),      // InEchoReps
+			in.Timestamp.Value(),      // InTimestamps
+			in.TimestampReply.Value(), // InTimestampReps
+			in.InfoRequest.Value(),    // InAddrMasks
+			in.InfoReply.Value(),      // InAddrMaskReps
+			0,                         // OutMsgs
+			Metrics.ICMP.V4PacketsReceived.Invalid.Value(), // OutErrors
+			out.DstUnreachable.Value(),                     // OutDestUnreachs
+			out.TimeExceeded.Value(),                       // OutTimeExcds
+			out.ParamProblem.Value(),                       // OutParmProbs
+			out.SrcQuench.Value(),                          // OutSrcQuenchs
+			out.Redirect.Value(),                           // OutRedirects
+			out.Echo.Value(),                               // OutEchos
+			out.EchoReply.Value(),                          // OutEchoReps
+			out.Timestamp.Value(),                          // OutTimestamps
+			out.TimestampReply.Value(),                     // OutTimestampReps
+			out.InfoRequest.Value(),                        // OutAddrMasks
+			out.InfoReply.Value(),                          // OutAddrMaskReps
+		}
+	case *inet.StatSNMPTCP:
+		tcp := Metrics.TCP
+		// RFC 2012 (updates 1213):  SNMPv2-MIB-TCP.
+		*stats = inet.StatSNMPTCP{
+			1,                                     // RtoAlgorithm
+			200,                                   // RtoMin
+			120000,                                // RtoMax
+			(1<<64 - 1),                           // MaxConn
+			tcp.ActiveConnectionOpenings.Value(),  // ActiveOpens
+			tcp.PassiveConnectionOpenings.Value(), // PassiveOpens
+			tcp.FailedConnectionAttempts.Value(),  // AttemptFails
+			tcp.EstablishedResets.Value(),         // EstabResets
+			tcp.CurrentEstablished.Value(),        // CurrEstab
+			tcp.ValidSegmentsReceived.Value(),   // InSegs
+			tcp.SegmentsSent.Value(),            // OutSegs
+			tcp.Retransmits.Value(),             // RetransSegs
+			tcp.InvalidSegmentsReceived.Value(), // InErrs
+			tcp.ResetsSent.Value(),              // OutRsts
+			tcp.ChecksumErrors.Value(),          // InCsumErrors
+		}
+	case *inet.StatSNMPUDP:
+		udp := Metrics.UDP
+		*stats = inet.StatSNMPUDP{
+			udp.PacketsReceived.Value(),   // InDatagrams
+			udp.UnknownPortErrors.Value(), // NoPorts
+			0, // InErrors
+			udp.PacketsSent.Value(),         // OutDatagrams
+			udp.ReceiveBufferErrors.Value(), // RcvbufErrors
+			0, // SndbufErrors
+			0, // InCsumErrors
+			0, // IgnoredMulti
+		}
+	default:
+		return syserr.ErrEndpointOperation.ToError()
```

Comment:
One implementation returns syserr.ErrEndpointOperation.ToError() and one returns syserror.EOPNOTSUPP. I feel like they should be consistent.

Reply:
If I understand it correctly, syserr.ErrEndpointOperation.ToError()  == syserror.EOPNOTSUPP, no?

Reply:
Yeah, it looks that way. Maybe return syserr.ErrEndpoint.ToError() in both implementations to be consistent?

Reply:
Added a separate comment.